jpa/keyset-pagination/boot-data-window-pagination/src/test/java/com/example/keysetpagination/web/controllers/AnimalControllerIT.java

```diff
@@ -89,22 +103,73 @@ void shouldSearchAnimals() throws Exception {
 
         CustomWindow<Map<String, String>> window = objectMapper.readValue(contentAsString, CustomWindow.class);
         List<Map<String, String>> animalResponses = window.getContent();
+        boolean windowLast = window.isLast();
         Map<String, String> animalResponsesLast = animalResponses.getLast();
         this.mockMvc
-                .perform(get("/api/animals/search")
+                .perform(post("/api/animals/search")
                         .param("pageSize", "2")
-                        .param("type", "Bird")
-                        .param("scrollId", String.valueOf(animalResponsesLast.get("id"))))
+                        .param("scrollId", String.valueOf(animalResponsesLast.get("id")))
+                        .content(
+                                """
+                                [
+                                  {
+                                    "queryOperator": "EQ",
+                                    "field": "type",
+                                    "values": [
+                                      "Bird"
+                                    ]
+                                  }
+                                ]
+                                """)
+                        .contentType(MediaType.APPLICATION_JSON))
                 .andExpect(status().isOk())
                 .andExpect(jsonPath("$.content.size()", is(1)))
                 .andExpect(jsonPath("$.content[0].type", is("Bird")))
                 .andExpect(jsonPath("$.last", is(true)));
     }
 
+    @Test
+    void shouldReturnResultForNotEqualType() throws Exception {
```

Comment:
@coderabbitai, can you please help me to write junits for all enum types of QueryOperator.

Reply:
@coderabbitai, can you please help me to write integration for all enum types of QueryOperator in AnimalControllerIT.java following existing patterns

Reply:
@coderabbitai, thank you for the help